react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+export default class Validate extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { password: '' };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    console.log('Password:', this.state.password);
+  }
+
+  render() {
+    const password = this.state.password;
+    if (password === '') {
+      return (
+        <div className="container">
+          <form onSubmit={this.handleSubmit}>
+            <label htmlFor="password"> Password: </label>
+            <div>
+              <input
+                id="password"
+                type="password"
+                value={this.state.password}
+                onChange={this.handleChange}
+              />
+              <i className="fas fa-times x-icon"></i>
+              <p className="error-message">Password is required</p>
+            </div>
+          </form>
+        </div>
+      );
+    } else if (password.length > 8) {
+      return (
+        <div className="container">
+          <form onSubmit={this.handleSubmit}>
+            <label htmlFor="password"> Password: </label>
+            <div>
+              <input
+                id="password"
+                type="password"
+                value={this.state.password}
+                onChange={this.handleChange}
+              />
+              <i className="fas fa-check check-icon"></i>
+            </div>
+          </form>
+        </div>
+      );
+    } else {
+      return (
+        <div className="container">
+          <form onSubmit={this.handleSubmit}>
+            <label htmlFor="password"> Password: </label>
+            <div>
+              <input
+                id="password"
+                type="password"
+                value={this.state.password}
+                onChange={this.handleChange}
+              />
+              <i className="fas fa-times x-icon"></i>
+              <p className="error-message">Your password is too short.</p>
+            </div>
+          </form>
+        </div>
```

Comment:
Instead of writing all of the elements all over for each of these, try using some variables (or better yet, some custom methods) to hold the parts that are different.